vllm/engine/arg_utils.py

```diff
@@ -552,23 +636,37 @@ def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
                                 **load_kwargs["pt_load_map_location"])
 
         # Guided decoding arguments
-        guided_decoding_kwargs = get_kwargs(DecodingConfig)
-        guided_decoding_group = parser.add_argument_group(
-            title="DecodingConfig",
-            description=DecodingConfig.__doc__,
+        structured_output_kwargs = get_kwargs(StructuredOutputConfig)
+        structured_output_group = parser.add_argument_group(
+            title="StructuredOutputConfig",
+            description=StructuredOutputConfig.__doc__,
+        )
+        structured_output_group.add_argument(
+            '--structured-output-config',
+            type=json.loads,
+            default=None,
+            help='The configurations for structured output config.'
+            ' Should be a JSON string.')
```

Comment:
CLI args for enture configs should go in the `vllm_group`, then it would be:

```suggestion
        vllm_group.add_argument(
            '--structured-output-config',
            **vllm_kwargs["structured_output_config"])
```